### DISCUSSION TITLE
Curious on where best to post process files for sonarr/radarr

### DISCUSSION THREAD

**Discussion by Milkbone911** [RandomNinjaAtk/arr-scripts#35] [Q&A] [answered]
Seeing that randomninja use to maintain a radarr with sma docker, to run the post processing in radarr and after switching from nzbget to sabnzbd with extended scripts has me wondering about where it would be best to run the post processing.

Previously after requesting a movie/tv show I would have the files available (assuming full download speed) in 20mins or less, with sabnzbd running sma that time has been extended. 99% of the time this isn't a big issues but everynow and then there are instances where having a quick turn around time is better. 

I've played with tdarr in the past using my server as well as the gpu on my main pc to do transcoding/standardizing of my media files. I know that tdarr will leave a copy of the media file in place and process after import so that the original download is available to be watched quickly and then the transcoded file will replace that once it has finished.

I was curious on the decision to run SMA in sabnzbd before import, and was wondering what benefits or drawbacks there is to using it in radarr/sonarr instead and using a similar process of leaving a copy in place and replacing once sma has finished, to take advantage of the quicker turn around/availability for those very limited times where it's wanted.

This is just some general questions and I'm not looking to solve any specific issues preventing my system from working. What are your thoughts on this and how you would best process files to make direct play/easiest streaming with out much of the, on the fly transcoding from plex to limit overhead on the cpu during a stream.

**Comment by RandomNinjaAtk** [comment, marked as answer]
I preprocess using sabnzbd and then allow it to import with no changes afterward, its my preferred workflow. I think its  a bit better because the media files are in their final format before the media clients pick them up. The post processing option means your modifying the files afterward, so file names that have the codecs and audio specs in the name may no longer match, the media clients will likely need to re-scan the file to detect the changes and etc...

Preprocessing avoids all of that non-sense... 

I also don't really transcode, I just repackage the files... The default SMA.ini config is what I personally use, I think your just better off sourcing material that is close to the desired format, rather than spending the time to transcode... If your going to transcode yourself, you should probably focus on remuxes as the source...

Anyways, just some thoughts...

Edit:
SMA is installed via the scripts to all of the containers because its used for various scripts that are available here, so you could still use it like you were previously with Radarr. I just don't use it in that way, only as a preprocess in sab and also with all the various video scripts that are included here...
